apps/user-portal/src/components/overview/widgets/account-status-widget.tsx

```diff
@@ -29,28 +33,236 @@ import { ThemeIcon } from "../../shared";
  */
 export const AccountStatusWidget: FunctionComponent<{}> = (): JSX.Element => {
     const { t } = useTranslation();
+    const profileCompletion: ProfileCompletion = useSelector((state: AppState) => state.profile.completion);
+
+    /**
+     * Get the profile status based on the profile completion percentage.
+     *
+     * @return {ProfileCompletionStatus}
+     */
+    const getProfileStatus = (): ProfileCompletionStatus => {
+
+        const percentage = getProfileCompletionPercentage();
+
+        if (percentage <= UIConstants.ERROR_ACCOUNT_STATUS_UPPER_LIMIT) {
+            return ProfileCompletionStatus.ERROR;
+        } else if (percentage <= UIConstants.WARNING_ACCOUNT_STATUS_UPPER_LIMIT) {
+            return ProfileCompletionStatus.WARNING;
+        }
+
+        return ProfileCompletionStatus.SUCCESS;
+    };
+
+    /**
+     * Return the profile completion percentage.
+     *
+     * @return {number}
+     */
+    const getProfileCompletionPercentage = (): number => {
+        return profileCompletion && profileCompletion.percentage ? profileCompletion.percentage : 0;
+    };
+
+    /**
+     * Get the completion percentage based on the completion status of
+     * required and optional fields.
+     *
+     * @param field - Field to check.
+     * @return {ProfileCompletionStatus}
+     */
+    const getFiledCompletionStatus = (field: any): ProfileCompletionStatus => {
```

Comment:
Shouldn't it be `getFieldCompletionStatus`?

Reply:
It has to be. I've fixed the typo.